detekt/wrapper/src/main/kotlin/io/buildfoundation/bazel/rulesdetekt/wrapper/SandboxExecutor.kt

```diff
@@ -0,0 +1,59 @@
+package io.buildfoundation.bazel.rulesdetekt.wrapper
+
+import java.io.File
+import java.io.FileOutputStream
+import java.io.PrintStream
+import java.nio.charset.Charset
+import java.util.concurrent.atomic.AtomicReference
+
+interface SandboxExecutor {
+
+    data class Result(val code: Int, val stdout: String, val stderr: String)
+
+    fun execute(arguments: Array<String>): Result
+
+    class Impl(private val executor: SandboxedExecutor) : SandboxExecutor {
+
+        override fun execute(arguments: Array<String>): Result {
+            val systemStdout = System.out
+            val systemStderr = System.err
+
+            val sandboxStdoutFile = File("sandbox-stdout.txt")
+            val sandboxStderrFile = File("sandbox-stderr.txt")
+
+            val result = AtomicReference<SandboxedExecutor.Result>()
+
+            PrintStream(FileOutputStream(sandboxStdoutFile).buffered()).use { sandboxStdout ->
+                PrintStream(FileOutputStream(sandboxStderrFile).buffered()).use { sandboxStderr ->
+                    System.setOut(sandboxStdout)
+                    System.setErr(sandboxStderr)
+
+                    result.set(executor.execute(arguments))
+
+                    listOf(sandboxStdout, sandboxStderr).forEach {
```

Comment:
At least make it an array!

Reply:
Removed the iteration, you allocation maniac.